src/library_wasmfs.js

```diff
@@ -213,14 +213,11 @@ FS.createPreloadedFile = FS_createPreloadedFile;
       assert(arguments[1] === undefined);
 #endif
 
-      return withStackSave(() => {
-        var pathBuffer = stringToUTF8OnStack(path);
-        var buf = __wasmfs_parse_path(pathBuffer);
-        assert(buf); // TODO: error handling
-        var ret = UTF8ToString(buf);
-        _free(buf);
-        return ret;
-      });
+      var buf = withStackSave(() => __wasmfs_parse_path(stringToUTF8OnStack(path)));
```

Comment:
I wonder if __wasmfs_parse_path could take an output buffer and assume it is at least PATH_MAX in size?  That would avoid the dynamic allocation here.

Reply:
PATH_MAX is the max overall path assumed by libc.  Since wasmfs sits below libs it seems reasonable to inherit the same limit.    This limit is for the combined path, not a single segment.

It looks like its currently set to 4096 in musl

Reply:
I don't recall the full topic, but I think @tlively explained to me that there was some difference between the path size supported by libc, and that supported by the actual underlying filesystem, which could be larger (and IIRC we decided on no explicit limit in WasmFS). I suggest for now we keep the allocation here, which is consistent with other functions that do the same, and we can discuss more when @tlively is back?

I think you're right on the combined path, though, I guess it's not per-segment.